Modules/_uuidmodule.c

```diff
@@ -91,45 +121,1730 @@ py_windows_has_stable_node(void)
+    self->bytes[15] = node;
+
+    return 0;
+}
+
+static PyObject *
+get_int(uuidobject *self)
+{
+    return _PyLong_FromByteArray((unsigned char *)self->bytes, 16, 0, 0);
+}
+
+static PyObject *
+get_uuidmod_attr(uuid_state *state, const char *name)
+{
+    PyObject *mod = PyImport_ImportModule("uuid");
+    if (mod == NULL) {
+        return NULL;
+    }
+
+    PyObject *ret = PyObject_GetAttrString(mod, name);
+    Py_DECREF(mod);
+    return ret;
+}
+
+static PyObject *
+get_SafeUUID(uuid_state *state)
+{
+    if (state->safe_uuid != NULL) {
+        return Py_NewRef(state->safe_uuid);
+    }
+
+    state->safe_uuid = get_uuidmod_attr(state, "SafeUUID");
+    if (state->safe_uuid == NULL) {
+        return NULL;
+    }
+    return Py_NewRef(state->safe_uuid);
+}
+
+static uuidobject *
+make_uuid(PyTypeObject *type)
+{
+    uuidobject *self = NULL;
+    uuid_state *state = get_uuid_state_by_cls(type);
+
+    Py_BEGIN_CRITICAL_SECTION(type);
+    if (state->freelist_size > 0) {
+        self = state->freelist;
+        state->freelist = (uuidobject *)self->weakreflist;
+        state->freelist_size--;
+    }
+    Py_END_CRITICAL_SECTION();
+
+    if (self != NULL) {
+        // Reinitialize the object from freelist
+        _Py_NewReference((PyObject *)self);
+    }
+    else {
+        self = PyObject_New(uuidobject, type);
+        if (self == NULL) {
+            return NULL;
+        }
+    }
+
+    self->is_safe = NULL;
+
+    self->weakreflist = NULL;
+    self->cached_hash = -1;
+
+    return self;
+}
+
+static PyObject *
+Uuid_alloc(PyTypeObject *type, PyObject *args, PyObject *kwargs)
+{
+    uuidobject *self = make_uuid(type);
+    if (self == NULL) {
+        return NULL;
+    }
+    memset(self->bytes, 0, 16);
+    return (PyObject *)self;
+}
+
+static void
+Uuid_dealloc(PyObject *obj)
+{
+    PyTypeObject *type = Py_TYPE(obj);
+    uuid_state *state = get_uuid_state_by_cls(type);
+
+    uuidobject *uuid = (uuidobject *)obj;
+    if (uuid->weakreflist != NULL) {
+        PyObject_ClearWeakRefs(obj);
+    }
+    Py_CLEAR(uuid->is_safe);
+
+    int skip_dealloc = 0;
+    if (type != state->UuidType) {
+        // We only apply the freelist optimization to the known C type,
+        // not any subclasses of it.
+        goto dealloc;
+    }
+
+    Py_BEGIN_CRITICAL_SECTION(type);
+    if (state->freelist_size < MAX_FREE_LIST_SIZE) {
+        uuidobject *head = state->freelist;
+        state->freelist = uuid;
+        uuid->weakreflist = (PyObject *)head;
+        state->freelist_size++;
+        skip_dealloc = 1;
+    }
+    Py_END_CRITICAL_SECTION();
+
+dealloc:
+    if (!skip_dealloc) {
+        type->tp_free(uuid);
+        // UUID is a heap allocated type so we have to decref the type ref
+        Py_DECREF(type);
+    }
+}
+
+
+static PyObject *
+Uuid_get_int(PyObject *o, void *closure)
+{
+    return get_int((uuidobject *)o);
+}
+
+static PyObject *
+Uuid_get_is_safe(PyObject *o, void *closure)
```

Comment:
Can we use `PyObject *self` and `uuidobject *u` instead? or `PyObject *op` instead? (I tried hard to have signatures that have meaningful names).